### PR TITLE
Config format: better support for whitespace in enumerations

### PR DESCRIPTION
**Pull request by vojtechtoman**
Changed the 'non-space-string' XSD type to collapse all
whitespace, and made sure that all enumerated types
extend from this type (some were extending xs:string).

Fixes https://github.com/hazelcast/hazelcast/issues/14919

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast-spring/src/main/resources/hazelcast-spring-4.0.xsd [hunk 1]
```diff
@@ -4149,40 +4150,29 @@
         </xs:sequence>
     </xs:complexType>
 
-    <xs:simpleType name="wan-ack-type-format-enum">
```
Comment (blazember) [inline hazelcast-spring/src/main/resources/hazelcast-spring-4.0.xsd]: Shouldn't we keep this (with `non-space-string` `base`) to keep the enumeration part of the validation?

Comment (mmedenjak) [reply hazelcast-spring/src/main/resources/hazelcast-spring-4.0.xsd]: Looks like this enum was moved from a XML node to the properties list so I don't know if we can enforce the enum in any way.

Comment (blazember) [reply hazelcast-spring/src/main/resources/hazelcast-spring-4.0.xsd]: Yeah, indeed, thanks. So it's a leftover to remove.

**Comment by mmedenjak** [comment]
Failed tests:
```
com.hazelcast.client.config.InvalidConfigurationClientTest.WhenInvalid_ListenerClassName
com.hazelcast.client.config.InvalidConfigurationClientTest.WhenInvalid_CredentialsClassName
```

**Comment by vojtechtoman** [comment]
run-lab-run

**Pull request merged**